pkg/clientgocred/BUILD

```diff
@@ -0,0 +1,35 @@
+package(default_visibility = ["//visibility:public"])
```

Comment:
packages that are only depended on by one binary should go in cmd/{thatbinary}/{pkg}

Reply:
We had an internal conversation on a bug, where a user wanted to build their own binary with the old Authprovider. 

This sparked the question, should we allow users to build their own binaries.Since, that question cannot actually be resolved quickly, I just moved the pkg folder outside to provide the option of building their own binary if needed. If any one reaches out to us, we can engage them toward building their own binaries.